lib/devices/ios/remote-debugger.js

```diff
@@ -148,9 +148,23 @@ RemoteDebugger.prototype.selectApp = function (appIdKey, cb, maxTries, tries) {
     // in iOS 8.2 the connect logic happens, but with an empty dictionary
     // which leads to the remote debugger getting disconnected, and into a loop
     if (_.isEmpty(pageDict)) {
-      var msg = "Empty page dictionary received. Trying again.";
-      return onRequiresRetry(this.appIdKey, msg);
+      return onRequiresRetry(this.appIdKey, "Empty page dictionary received. Trying again.");
     }
+
+    if (currentUrl) {
```

Comment:
what if currentUrl is not truthy? seems like this will loop infinitely since found is only set to true in this block. if currentUrl is required, we should `assert.ok(currentUrl)` above and get rid of this `if` statement.

unless i'm reading this wrong